Sources/Request.swift

```diff
@@ -4,14 +4,17 @@ public struct Request: Message {
     public var version: Version
     public var headers: Headers
     public var body: Body
-    public var storage: [String: Any] = [:]
+    public var cookies: [String: String]
```

Comment:
I think `cookies` shouldn't be a stored property on the `Request` only in the `Response`. Explaining briefly.. all headers can be represented with `headers: [CaseInsensitiveString: String]` (wrapped inside `Headers` struct, except `Set-Cookie`. `Set-Cookie` is the only HTTP header that can appear more than once. This is why it's necessary to have a separate stored property for it.  `cookies: Set<String>`  or the more high level `cookies: Set<Cookie>`.

Because all other headers, in this case specifically, `Cookie` header in the `Request`, can be stored in `headers`. I think we shouldn't have a stored property for `cookies` in `Request`. `Set-Cookie` header in `Response` is a special case in the RFC. `Cookie` header in `Request` is not a special case. Because of this I think the only header that should be a stored property is the only one that needs to be a stored property.